src/Workspaces/Core/Portable/CodeActions/CodeAction.cs

```diff
@@ -65,6 +65,11 @@ public abstract class CodeAction
         internal virtual ImmutableArray<CodeAction> NestedCodeActions
             => ImmutableArray<CodeAction>.Empty;
 
+        /// <summary>
+        /// Gets custom tags for the CodeAction.
+        /// </summary>
+        public virtual ImmutableArray<string> CustomTags => ImmutableArray<string>.Empty;
```

Comment:
does this need to be public?

Reply:
I think in the design meeting we decided that we will expose this as a public API, is that correct @jasonmalinowski? Also, if we are exposing this public property, we should also expose new `Code.Create` overloads that take custom tags argument.

Reply:
It looks like we discussed this on the 12/7/2020 design meeting; I don't recall the details (and the conclusion from the written notes isn't clear), but my memory matches @mavasani's in that we were OK making a public API like this. Especially since we already have one on Diagnostic and have good experience with it.